corehq/apps/export/views.py

```diff
@@ -403,7 +403,20 @@ def prepare_custom_export(self, in_data):
             export_filters, export_specs = self._process_filters_and_specs(in_data)
             export_instances = [self._get_export(self.domain, spec['export_id']) for spec in export_specs]
             self._check_deid_permissions(export_instances)
-            self._check_export_size(export_instances, export_filters)
+
+            # Check export isn't too big to download
+            count = 0
+            for instance in export_instances:
+                count += get_export_size(instance, export_filters)
+            if count > MAX_EXPORTABLE_ROWS and not PAGINATED_EXPORTS.enabled(self.domain):
+                raise ExportAsyncException(
+                    _("This export contains %(row_count)s rows. Please change the "
+                      "filters to be less than %(max_rows)s rows.") % {
+                        'row_count': count,
+                        'max_rows': MAX_EXPORTABLE_ROWS
+                    }
+                )
+
```

Comment:
Same here - I prefer the abstraction.  You can read `_check_export_size` and know what it does without needing to care about the implementation.

Reply:
https://github.com/dimagi/commcare-hq/commit/3b3a76508989fd1b6b21445255ba45de4a7fbdef